homeassistant/components/blink/config_flow.py

```diff
@@ -94,23 +91,36 @@ async def async_step_user(self, user_input=None):
 
     async def async_step_2fa(self, user_input=None):
         """Handle 2FA step."""
+        errors = {}
         if user_input is not None:
             pin = user_input.get(CONF_PIN)
-            if await self.hass.async_add_executor_job(
-                self.blink.login_handler.send_auth_key, self.blink, pin
-            ):
-                return await self.async_step_user(user_input=self.data)
+            try:
+                if await self.hass.async_add_executor_job(
+                    _send_blink_2fa_pin, self.auth, pin
+                ):
+                    return await self.async_step_finish()
+                errors["base"] = "invalid_access_token"
+            except BlinkSetupError:
+                errors["base"] = "cannot_connect"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
```

Comment:
We prefer only wrapping the line that can raise with try... except.
```suggestion
            try:
                valid_token = await self.hass.async_add_executor_job(
                    _send_blink_2fa_pin, self.auth, pin
                )
            except BlinkSetupError:
                errors["base"] = "cannot_connect"
            except Exception:  # pylint: disable=broad-except
                _LOGGER.exception("Unexpected exception")
                errors["base"] = "unknown"
            else:
                if valid_token:
                    return self._async_finish_flow()
                errors["base"] = "invalid_access_token"
```